citest/reporting/simplify_entity_transforms.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helper classes and methods for special case interpreting Snapshot Entities.
+
+These classes rewrite some of the entities to simplify their reporting
+interpretations to provide a more concise data model.
+"""
+
+class EdgeLabelValueTransformer(object):
+  """Transforms a snapshot entity edge label and value depending on context.
+
+
+  The instance is bound to the original entity and is called with each edge
+  to extract the label and value for that edge giving the instance a chance
+  to rewrite the label and/or value, or to ignore it completely.
+  """
+
+  @property
+  def entity(self):
+    """The SnapshotEntity dictionary bound at construction."""
+    return self.__entity
+
+  @property
+  def entity_manager(self):
+    """The EntityManager bound at construction."""
+    return self.__entity_manager
+
+  @staticmethod
+  def _find_edge(entity, name):
+    """Return the edge with the given label value.
+
+    Args:
+      entity: [dict] The snapshot entity dict
+      name: [string] The label name we're looking for.
+    """
+    if entity is None:
+      return None
+    for edge in entity.get('_edges', []):
+      if edge.get('label') == name:
+        return edge
+    return None
+
+  def __init__(self, entity, entity_manager):
+    """Constructor."""
+    self.__entity = entity
+    self.__entity_manager = entity_manager
+
+  def __call__(self, edge):
+    """Returns label, value for the given edge.
+
+    Args:
+      edge: [dict] The edge dictionary
+
+    Returns:
+      label, value tuple where label is None means ignore the edge.
+    """
+    label = edge.get('label', '?unlabeled')
```

Comment:
Edges should always have a label to name the relationship to the value. However this isnt strictly enforced in the writer, and this library is decoupled from the data anyway so I need to accomodate errors. Giving it a visible name hints the fact that it wasnt named, which should be addressed. This much isnt a change, I just moved it from being hardcoded in html_renderer.py to being the default here when html_renderer runs through the transform. What's new is that I now use None to mean ignore the value altoghether. Since the source cannot contain None (we give None values names here), this is unambiguous.